runtime/src/group.rs

```diff
@@ -0,0 +1,392 @@
+//! # Group Module
+//!
+//! The Group module is used to manage a set of identities. A group of identities can be a
+//! collection of KYC providers, council members for governance and so on. This is an instantiable
+//! module.
+//!
+//! ## Overview
+//! Allows control of membership of a set of `IdentityId`s, useful for managing membership of a
+//! collective.
+//!
+//! - Add a new identity
+//! - Remove identity from the group
+//! - Swam members
+//! - Reset group members
+//!
+//! ### Dispatchable Functions
+//!
+//! - `add_member` - Adds a new identity to the group.
+//! - `remove_member` - Remove identity from the group if it exists.
+//! - `swap_member` - Replace one identity with the other.
+//! - `reset_members` - Re-initialize group members.
+//!
+use codec::{Decode, Encode, HasCompact};
+use primitives::IdentityId;
+use rstd::prelude::*;
+use sr_primitives::{traits::EnsureOrigin, weights::SimpleDispatchInfo};
+use srml_support::{decl_event, decl_module, decl_storage};
+use system::ensure_root;
+
+pub trait Trait<I = DefaultInstance>: system::Trait {
+    /// The overarching event type.
+    type Event: From<Event<Self, I>> + Into<<Self as system::Trait>::Event>;
+
+    /// Required origin for adding a member (though can always be Root).
+    type AddOrigin: EnsureOrigin<Self::Origin>;
+
+    /// Required origin for removing a member (though can always be Root).
+    type RemoveOrigin: EnsureOrigin<Self::Origin>;
+
+    /// Required origin for adding and removing a member in a single action.
+    type SwapOrigin: EnsureOrigin<Self::Origin>;
+
+    /// Required origin for resetting membership.
+    type ResetOrigin: EnsureOrigin<Self::Origin>;
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait<I>, I: Instance=DefaultInstance> as Group {
+        /// Identities that are part of this group
+        Members get(members) config(): Vec<IdentityId>;
+    }
+    add_extra_genesis {
+        config(phantom): rstd::marker::PhantomData<(T, I)>;
+    }
+}
+
+decl_event!(
+	pub enum Event<T, I=DefaultInstance> where
+		<T as system::Trait>::AccountId,
+		<T as Trait<I>>::Event,
+	{
+		/// The given member was added; see the transaction for who.
+		MemberAdded,
```

Comment:
Events should include the identity of member affected

Reply:
Makes sense. Latest commit should reflect that.